building_sim_plugins/building_plugins_common/src/slotcar_common.cpp

```diff
@@ -296,7 +296,11 @@ bool SlotcarCommon::emergency_stop(
   bool need_to_stop = false;
   for (const auto& obstacle_pos : obstacle_positions)
   {
-    if ((obstacle_pos - stop_zone).norm() < _stop_radius)
+    Eigen::Vector3d detect_vector = obstacle_pos - stop_zone;
+    // check whether the obstacle model is in the same level with the slotcar, hardcode adjacent level elevation 2.0 m
+    if (std::abs(detect_vector(2)) > 2.0) continue;
```

Comment:
Is the level elevation in traffic editor a fixed parameter? I wonder if there is a way to make this value a variable that is fetched from the world settings?

Reply:
Yea but if level differences can be changed this could still lead to unpredictable behavior. For example if there was a building with very low levels (let's say 1.5m) the slotcar would stop for objects that are upstairs, if the levels were very high and there was a very tall object that is over 2 meters (i.e. a tall pillar in a warehouse) the slotcar would just go through it.

Reply:
This issue occurs because the common actors I used in crowd simulation has a z=1.0 (on the waist of the human body), and currently the slot car is using 0.75 as the stop_range, so there is no way the e-stop will be triggered. I don't know if there are already designed human models can be used (with base position z = 0). If there are, I can try for that. 
Now the tall pillar situation:
the `stop_zone` point is at the same level with the slot car (z_slotcar = z_level_elevation)
the `obstacle_pos` is the designed pose of the object (x_pillar, y_pillar, z_pillar). 
* I think we should talk about the pillar will not go through all levels right? like one pillar should remain within one level, this pillar should not pass through several levels. 
if z_pillar = z_level_elevation, then, z_pillar = z_slotcar, definitely will trigger the e-stop.
if z_pillar is in the middle of the pillar model, then z_level_elevation < z_pillar < z_level_elevation + 2.0, this will be captured by my fixed code.
* If the pillar pass through multiple levels, then the e-stop trigger might be random at different level.

Reply:
OK, sure.